_profiles/drupal/resource-versioning/index.md

```diff
@@ -0,0 +1,273 @@
+    context resource object.
+  - `successor-version`: links to the version which immediately succeeded the
+    context resource object.
+  - `prior-working-copy`: links to the working copy which immediately preceded
+    the context resource object.
+  - `subsequent-working-copy`: links to the working copy which immediately
+    preceded the context resource object.
+
+A server **MAY** provide an array of any of these links to support branching.
+
+For example, in the following version history:
+
+```
+    _c_
+   /   \
+  /__b__d__   __f__   __h
+ /         \ /     \ /
+a           e       g
+```
+
+`g` is the latest version. Both `a` and `e` were previously the latest version.
+No other revisions were ever the latest version. In this example, the following
+links could be provided:
+
+| Revision | `latest-version` | `working-copy` | `predecessor-version` | `successor-version` | `prior-working-copy` | `subsequent-working-copy` |
+| :------: | :--------------: | :------------: | :-------------------: | :-----------------: | :------------------: | :-----------------------: |
+| `a`      | `g`              | `h`            | no link               | `e`                 | no link              | `b`, `c`                  |
+| `b`      | `g`              | `h`            | `a`                   | `e`                 | `a`                  | `c`                       |
+| `c`      | `g`              | `h`            | `a`                   | `e`                 | `b`                  | `d`                       |
+| `d`      | `g`              | `h`            | `a`                   | `e`                 | `b`, `c`             | `e`                       |
+| `e`      | `g`              | `h`            | `a`                   | `g`                 | `d`                  | `f`                       |
+| `f`      | `g`              | `h`            | `e`                   | `g`                 | `e`                  | `g`                       |
+| `g`      | no link          | `h`            | `e`                   | no link             | `f`                  | `h`                       |
+| `h`      | `g`              | no link        | `g`                   | no link             | `g`                  | no link                   |
+
+> Note: In this example, `f` has both a `predecessor-version` and
+> `prior-working-copy` link to `e` because `e` was a version and `e` was also the
+> revision to which changes could be applied prior to `f`'s creation.
+
+# Version History
+
+A server **MAY** provide a "version history" endpoint. The primary data
+of a response document from a version history endpoint must be a collection of
+resource objects.
+
+Unless an `id` contains version information, the `type` and `id` members of each
+resource object in the collection **MUST** be the same.
+
+If provided, resource objects' `self` links **MUST NOT** be the same.
+
+# Version Negotiators
```

Comment:
[Design feedback/interoperable implementations review requirement] If I'm understanding correctly, each server implementation can define its own version negotiators, with multiple servers using the same negotiator name in different ways (and no central registry). That seems not great for interoperability. It would also seem to mean that no other version negotiators can ever be standardized as part of this profile, because they could conflict with existing implementations. Is that really what you want?

My gut instinct would be to make this profile's spec the canonical list of all legal version negotiators. Then, you can add more over time to this spec as they're requested. To support people who really need to use a version negotiator that isn't part of your profile, you could have some set of negotiator names/schemes/namespaces that are allowed to have implementation-specific meanings.

Reply:
I guess I wanted to discourage non-standard negotiators being used as a standard and keep them isolated. My feeling was that sending an email saying "hey, I'd like to add a negotiator like this..." doesn't seem like a super high barrier. In some ways, it's even less difficult than opening a PR.

What if we add this:

> Note: Don't be shy! New version negotiators are more than welcome, the editors want to see this profile proliferate and that means accepting versioning strategies like yours!

Reply:
@e0ipso, would that work?